app/main/routes.py

```diff
@@ -121,20 +117,21 @@ def download():
                 "request_type": "download",
             },
         )
-        return send_file(
-            file_content,
-            download_name=f"download-{current_datetime}.{file_format}",
-            as_attachment=True,
-            mimetype=content_type,
-        )
+
+        if status_code == 204:
+            return redirect(url_for("main.request_received"))
+        else:
+            current_app.logger.error(
+                "Response status code from data-store/trigger_async_download: {content_type}",
+                extra=dict(content_type=status_code),
+            )
+            return render_template("500.html")
 
 
-@bp.route("/request-received", methods=["GET", "POST"])
+@bp.route("/request-received", methods=["GET"])
 @login_required(return_app=SupportedApp.POST_AWARD_FRONTEND)
 def request_received():
-    context = {
-        "user_email": g.user.email,
-    }
+    context = {"user_email": g.user.email}
```

Comment:
Is it maybe overly complex to pass through a single value as a dictionary. Could we just pass through `user_email` directly?

Reply:
Sam said to not change it now, to avoid conflicts between Israr's branch and this one. (please see his 3rd comment up from here).